firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -112,14 +128,87 @@ public void onFailure(@NonNull @NotNull Exception e) {
     return taskCompletionSource.getTask();
   }
 
+  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
+  @NonNull
+  public Task<Void> signInTester() {
+    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
+      signInCancellationSource.cancel();
+    }
+
+    signInCancellationSource = new CancellationTokenSource();
+    signInTaskCompletionSource = new TaskCompletionSource<>(signInCancellationSource.getToken());
+
+    Context context = firebaseApp.getApplicationContext();
+    AlertDialog alertDialog = getSignInAlertDialog(context);
+    alertDialog.show();
+
+    return signInTaskCompletionSource.getTask();
+  }
+
   /**
    * Returns an AppDistributionRelease if one is available for the current signed in tester. If no
    * update is found, returns null. If tester is not signed in, presents the tester with a Google
    * sign in UI
    */
   @NonNull
   public Task<AppDistributionRelease> checkForUpdate() {
-    return Tasks.forResult(null);
+
+    if (checkForUpdateTaskCompletionSource != null
+        && !checkForUpdateTaskCompletionSource.getTask().isComplete()) {
+      checkForUpdateCancellationSource.cancel();
+    }
+
+    checkForUpdateCancellationSource = new CancellationTokenSource();
+    checkForUpdateTaskCompletionSource =
+        new TaskCompletionSource<>(checkForUpdateCancellationSource.getToken());
+
+    firebaseInstallationsApi
+        .getId()
```

Comment:
This makes a network call to get the FID. Should this be in a background thread? I don't know the answer to this, curious what you think

Reply:
Since `getId` returns a task, it should not be blocking. I think this should be fine